modules/registrars/openprovider/openprovider.php

```diff
@@ -329,6 +330,49 @@ function openprovider_ResendIRTPVerificationEmail(array $params)
     return openprovider_registrar_launch_decorator('resendIRTPVerificationEmail', $params);
 }
 
+
+function openprovider_config_validate($params)
+{
+    $username = $params['Username'];
+    $password = $params['Password'];
+    $testMode = $params['test_mode'];
+    $resourcePath = '/v1beta/auth/login'; //Resource path to login API
+
+    $valid = false;
+    $baseUrl = Configuration::get('api_url');
+    $env = 'Production';
+
+    if ($testMode == 'on') {
+        $env = 'Sandbox';
+        $baseUrl = Configuration::get('api_url_cte');
+    }
+
+    $url = "{$baseUrl}{$resourcePath}"; 
+
+    $data = array(
+        "username" => $username,
+        "password" => $password,
+        "ip" => "0.0.0.0",
+    );
+
+    $encodedData = json_encode($data);
+    $curl = curl_init($url);
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($curl, CURLOPT_HTTPHEADER, array('Content-Type:application/json'));
+    curl_setopt($curl, CURLOPT_POST, true);
+    curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, false);
+    curl_setopt($curl, CURLOPT_POSTFIELDS, $encodedData);
+    $result = curl_exec($curl);
+    curl_close($curl);
+    $response = json_decode($result);
+    if ($response->data->token && $response->data->reseller_id) {
+        $valid = true;
```

Comment:
you can remove this $valid variable and check directly on these values

`if (!$response->data->token || ! $response->data->reseller_id) {
  throw Error...
  }

Reply:
This change has been added